src/main/java/org/jabref/gui/bibtexextractor/ExtractBibtexDialog.java

```diff
@@ -46,7 +46,18 @@ public ExtractBibtexDialog() {
 
         buttonParse = (Button) getDialogPane().lookupButton(parseButtonType);
         buttonParse.setTooltip(new Tooltip((Localization.lang("Starts the extraction and adds the resulting entries to the currently opened database"))));
-        buttonParse.setOnAction(event -> viewModel.startParsing());
+        buttonParse.setOnAction(event -> {
+            if (!preferencesService.getImportSettingsPreferences().isGrobidEnabled() && !preferencesService.getImportSettingsPreferences().isGrobidOptOut()) {
+                boolean confirmGrobidUsage = dialogService.showConfirmationDialogWithOptOutAndWait(
+                        "Remote services",
+                        "Allow sending PDF files and raw citation strings to a JabRef online service (Grobid) to determine Metadata",
```

Comment:
"Allow sending PDF files and raw citation strings to a JabRef online service (Grobid) to determine Metadata. Usually provides better results",


I would add this produces better results.
Use JabRef's heuristic parser. Results might be worse.